cloudflare/resource_cloudflare_certificate_pack.go

```diff
@@ -62,6 +63,32 @@ func resourceCloudflareCertificatePackCreate(d *schema.ResourceData, meta interf
 		certificatePackID = certPackResponse.ID
 	}
 
+	if d.Get("wait_for_active_status").(bool) {
+		return resource.Retry(d.Timeout(schema.TimeoutCreate)-time.Minute, func() *resource.RetryError {
+			certificatePack, err := client.CertificatePack(context.Background(), zoneID, certificatePackID)
+			if err != nil {
+				return resource.NonRetryableError(errors.Wrap(err, "failed to fetch certificate pack"))
+			}
+			if len(certificatePack.Certificates) == 0 {
+				return resource.RetryableError(fmt.Errorf("certificate list in response is empty"))
+			}
+			for _, certificate := range certificatePack.Certificates {
+				if certificate.Status != "active" {
+					return resource.RetryableError(fmt.Errorf("expected all certificates in certificate pack to be active state but certificate %s was in state %s", certificate.ID, certificate.Status))
+				}
+			}
+
+			// if we get here, the new cert pack is active
```

Comment:
is there a reason we're duplicating the `d.SetId` and `Read` operations here? we should be able to reuse what we already have a few lines further down.

Reply:
No good reason, I had rushed this out so just hadn't noticed the duplication. Should be resolved in latest commits.

Reply:
No good reason, I had rushed this out so just hadn't noticed the duplication. Should be resolved in latest commits.